### ISSUE TITLE
Multiple triggers on same object

### ISSUE THREAD

**Issue by cmcfarling** [afawcett/declarative-lookup-rollup-summaries#640]
DLRS is being used to concatenate values from the task object and update a field on a custom object. The values being pulled from the task are being generated by a different task trigger. DLRS is set to use Realtime Calculation Mode

The custom task trigger populates a field on the task with the name of the opportunity the task is associated with. Those values are then concatenated and dumped into a field on a custom object that the tasks are also associated with.

This mostly works but occasionally the custom object doesn't update. I assume this is due to a situation where the DLRS trigger fires before the custom trigger. Besides switching from realtime calc mode to running on a schedule, I wondering of there are any suggestions to get around the issue.

Another thought, what if I kept the realtime summary in place and set up a second scheduled summary that did the same actions. It could just run once a day or so to catch any records that happened to get missed by the realtime summary. Is that doable?

**Comment by cmcfarling** [comment]
Ok, I understand the concept but coding it looks to be beyond my ability. I looked through the documentation I could find on the Developer API but I'm not exactly sure what to include where. Below is the existing custom trigger. Is it just a matter of adding a few lines of code to this? Thanks for any input.

```
/**
* Description : It calculate the formula for Opp name
*/
public class TaskTriggerHandler extends TriggerHandler{
    List<Id> oppIdList = new List<Id>();
    Map<Id,String> oppNameMap = new Map<Id,String>();
    Map<Id,Id> taskConsultantIdMap = new Map<Id,Id>();
    Map<Id, String> taskIdOppNameMap = new Map<Id, String>();    
    
    /**
    * Date   : 30/3/2017
    * Description : This is before insert trigger handler method
    * Param : Void
    * Return : Void 
    */
    public override void beforeInsert() {
        getOppIds();
        getOppNames();
        setOppNames();
    }
    
    
    /**
    * Date   : 30/3/2017
    * Description : This is before update trigger handler method
    * Param : 
    * Return : 
    */
    public override void beforeUpdate(){
        getOppIds();
        getOppNames();
        setOppNames();
        getConsultantsListUpdated();
    }
    
    /**
    * Date   : 06/08/2013
    * Description : Gets the Opp ids
    * Param : 
    * Return : 
    */
    public void getOppIds(){
        for(Task tsk : (List<Task>)Trigger.new){
            System.debug('===tsk===>'+tsk);
            if((''+tsk.WhatId).startsWith('006')){ //&& tsk.Activity_Type__c == 'Deployment' && tsk.Status == 'In Progress'
                oppIdList.add(tsk.WhatId);
            }
        }
    }
    
    
    public override void afterUpdate() {
        for(Task tsk : (List<Task>)Trigger.new){
            if(tsk.Status == 'Completed' && tsk.Consultant__c != null){
                taskConsultantIdMap.put(tsk.Consultant__c, tsk.Id);
                taskIdOppNameMap.put(tsk.Id, tsk.Opportunity_Name__c);
            }
        }
        getConsultantsListUpdated();
    }
    
    public void getConsultantsListUpdated(){
        if(taskConsultantIdMap.size() > 0){
            List<Consultant__c> consultantList = [SELECT Id, Current_Assignment2__c FROM Consultant__c WHERE Id IN :taskConsultantIdMap.keySet() AND Current_Assignment2__c != null];
            for(Consultant__c consultant : consultantList){
                String currentlyAssignedConsultant = '';
                for(String cName : consultant.Current_Assignment2__c.split(';')){
                    if(cName != taskIdOppNameMap.get(taskConsultantIdMap.get(consultant.Id))){
                        currentlyAssignedConsultant += cName + ';';
                    }
                    consultant.Current_Assignment2__c = currentlyAssignedConsultant.removeEnd(';');
                }
            }
            if(consultantList.size() > 0 ){
                UPDATE consultantList; 
            }
        }
    }
    
    /**
    * Date   : 06/08/2013
    * Description : It sets the opp ids
    * Param : 
    * Return : 
    */
    public void setOppNames(){
        for(Task tsk : (List<Task>)Trigger.new){
            if((''+tsk.WhatId).startsWith('006') && oppNameMap.containsKey(tsk.WhatId)){
                tsk.Opportunity_Name__c = oppNameMap.get(tsk.WhatId);
            }
        }
    }
    
    /**
    * Date   : 06/08/2013
    * Description : It gets the opp names
    * Param : 
    * Return : 
    */
    public void getOppNames(){
        for(Opportunity opp : [SELECT Id, Name FROM Opportunity WHERE Id IN : oppIdList]){
            oppNameMap.put(opp.Id, opp.Name);
        }
    }
}

```

**Comment by afawcett** [comment]
Add the following line...

    dlrs.RollupService.rollup(Trigger.oldMap, Trigger.newMap, Task.SObjectType)

To the **afterUpdate**, **afterDelete** and **afterInsert** methods. Make sure to remove the dlrs trigger via Manage CHild Trigger and set your Rollup type to Developer.

~~~
public override void afterUpdate() {
    dlrs.RollupService.rollup(Trigger.oldMap, Trigger.newMap, Task.SObjectType)
    // Rest of your code...
}
public override void afterInsert() {
    dlrs.RollupService.rollup(Trigger.oldMap, Trigger.newMap, Task.SObjectType)
    // Rest of your code...
}
public override void afterDelete() {
    dlrs.RollupService.rollup(Trigger.oldMap, Trigger.newMap, Task.SObjectType)
    // Rest of your code...
}
~~~

**Comment by cmcfarling** [comment]
Thanks for clarifying. The existing trigger has been updated to include the DLRS code for each of the three event handlers. There is a different issue now. I have 2 DLRS summaries to roll up data from the Task object to a custom object (Consultant__c). One of them works, one does not. They're both very similar. One concatenates text and one counts records.
ConsultantCurrentAssignment is working
ConsultantDeploymentCount in not working

I've tried enabling just ConsultantDeploymentCount by itself, checked field permissions, viewed debug logs but can't determine what the issue is.

![consultantcurrentassignment](https://user-images.githubusercontent.com/36546584/37353861-aa8d5740-26b6-11e8-8834-a48f177acd46.PNG)


![consultantdeploymentcount](https://user-images.githubusercontent.com/36546584/37353866-ad7f23f2-26b6-11e8-8e58-ec48e4c9a44d.PNG)
